processor/batchprocessor/batch_processor.go

```diff
@@ -174,32 +172,37 @@ func (bp *batchProcessor) resetTimer() {
 
 func (bp *batchProcessor) sendItems(triggerMeasure *stats.Int64Measure) {
 	// Add that it came form the trace pipeline?
-	stats.Record(bp.exportCtx, triggerMeasure.M(1), statBatchSendSize.M(int64(bp.batch.itemCount())))
+	stats.RecordWithTags(bp.exportCtx, bp.mutators, triggerMeasure.M(1), statBatchSendSize.M(int64(bp.batch.itemCount())))
 
 	if bp.telemetryLevel == configtelemetry.LevelDetailed {
-		stats.Record(bp.exportCtx, statBatchSendSizeBytes.M(int64(bp.batch.size())))
+		stats.RecordWithTags(bp.exportCtx, bp.mutators, statBatchSendSizeBytes.M(int64(bp.batch.size())))
 	}
 
 	if err := bp.batch.export(bp.exportCtx, bp.sendBatchMaxSize); err != nil {
 		bp.logger.Warn("Sender failed", zap.Error(err))
 	}
+	// Reset the context
+	bp.exportCtx = context.Background()
```

Comment:
Would you mind expanding upon why it is important to reset the context here?

Reply:
I am not sure if a single instance of a batch processor can be used by multiple different routines but if it can then this would introduce a race condition.

Reply:
added an expanded comment.  It is only used by a single routine.